internal/common/conversion/schema_generation.go

```diff
@@ -11,14 +11,44 @@ import (
 func DataSourceSchemaFromResource(rs schema.Schema, requiredFields []string, overridenFields map[string]dsschema.Attribute) dsschema.Schema {
 	blocks := convertBlocks(rs.Blocks, requiredFields)
 	attrs := convertAttrs(rs.Attributes, requiredFields)
-	for name, attr := range overridenFields {
-		if attr == nil {
-			delete(attrs, name)
-		} else {
-			attrs[name] = attr
+	overrideFields(attrs, overridenFields)
+	ds := dsschema.Schema{Attributes: attrs, Blocks: blocks}
+	UpdateSchemaDescription(&ds)
+	return ds
+}
+
+func PluralDataSourceSchemaFromResource(rs schema.Schema, requiredFields []string, overridenFields, overridenRootFields map[string]dsschema.Attribute, overrideResultsDoc string, hasLegacyFields bool) dsschema.Schema {
+	blocks := convertBlocks(rs.Blocks, nil)
+	if len(blocks) > 0 {
+		panic("blocks not supported yet in auto-generated plural data source schema as they can't go in ListNestedAttribute")
+	}
+	attrs := convertAttrs(rs.Attributes, nil)
+	overrideFields(attrs, overridenFields)
+	rootAttrs := convertAttrs(rs.Attributes, requiredFields)
+	for name := range rootAttrs {
+		if !slices.Contains(requiredFields, name) {
+			delete(rootAttrs, name)
 		}
 	}
-	ds := dsschema.Schema{Attributes: attrs, Blocks: blocks}
+	overrideFields(rootAttrs, overridenRootFields)
+	resultsDoc := "List of returned documents that MongoDB Cloud provides when completing this request."
+	if overrideResultsDoc != "" {
+		resultsDoc = overrideResultsDoc
+	}
+	rootAttrs["results"] = dsschema.ListNestedAttribute{
+		Computed: true,
+		NestedObject: dsschema.NestedAttributeObject{
+			Attributes: attrs,
+		},
+		MarkdownDescription: resultsDoc,
+	}
+	if hasLegacyFields {
+		rootAttrs["id"] = dsschema.StringAttribute{Computed: true}
```

Comment:
why is this necessary?

Reply:
it's needed for streamconnection and streaminstance, and maybe more if we do the migration to TPF

Reply:
I think this was a limitation of SDKv2, however, expecting new resources to not wanting this low level pagination args. So, ok to keep the `id`

Reply:
yeah, i added "legacy" in the param name to make it clear that we should try to avoid using it